### PR TITLE
[BugFix] Listening for new clean up action 

### PR DESCRIPTION
**Pull request by JoshuaLai**
## Purpose
New `CleanupCallStates` action that will reset the the recording and transcript value. 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
User A and User B join teams call
User A begins recording and transcription
User B loses internet and is kicked out into the set up view
User A stops the recording and transcription
User B turns on internet and rejoins call
Verify the banner is not showing in call

### CODE REVIEW

**Review by palatter — COMMENTED** [review]
Looks good and tested.

**Review by palatter — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/Manager/CompositeErrorManager.swift [hunk 1]
```diff
@@ -33,8 +33,10 @@ class CompositeErrorManager: ErrorManager {
             case .fatal:
                 update(error: error)
                 respondToFatalError(code: error.code)
+                store.dispatch(action: CallingAction.CleanupCallStates())
             case .callState:
                 update(error: error)
+                store.dispatch(action: CallingAction.CleanupCallStates())
```
Comment (kevinyulu) [inline AzureCommunicationUI/AzureCommunicationUI/Presentation/Manager/CompositeErrorManager.swift:39]: I tested the iOS app by following the repro step in the PR. The UX looks good. One observation is that when the call has an error, this `case` is triggered many times. By adding dispatch here will make app dispatch many `CallingAction.CleanupCallStates()` actions as well. Maybe there is a loop going on?

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUI/Presentation/Manager/CompositeErrorManager.swift:39]: This is an interesting find. I will take a look into it. If we are dispatching the same error multiple times we may need a check in here. Did this happen when you were disconnecting from the internet?

Comment (kevinyulu) [reply AzureCommunicationUI/AzureCommunicationUI/Presentation/Manager/CompositeErrorManager.swift:39]: > Did this happen when you were disconnecting from the internet?

@JoshuaLai I remember I started to see the dispatches as soon as the call was disconnected do to Internet error. (UX: as soon as I was landed back on the setup screen)

File: AzureCommunicationUI/AzureCommunicationUI/Redux/Action/CallingAction.swift [hunk 2]
```diff
@@ -25,6 +25,8 @@ struct CallingAction {
     struct TranscriptionStateUpdated: Action {
         let isTranscriptionActive: Bool
     }
+
+    struct CleanupCallStates: Action {}
```
Comment (jimchou-dev) [inline AzureCommunicationUI/AzureCommunicationUI/Redux/Action/CallingAction.swift:29]: Should there be unit test for this action?

Comment (JoshuaLai) [reply AzureCommunicationUI/AzureCommunicationUI/Redux/Action/CallingAction.swift:29]: Yes thats a good point, I need to run this by Wayne and Inderpal as well. So I will introduce the unit tests after that conversation.

**Comment by JoshuaLai** [comment]
Discussed a new plan for how to clean up the call states and error object.

**Pull request closed without merging**